tools/azsdk-cli/specs/1-env-setup-env-verification.spec.md

```diff
@@ -0,0 +1,581 @@
+  - Improve output formatting (colors, icons, clear sections)
+  - Add documentation to README
+  - Create examples and troubleshooting guide
+  - Add copilot-instructions examples for agent usage
+
+### Phase 4 (Future): Dynamic Discovery
+
+- **Milestone**: Requirements derived from source of truth config
+- **Timeline**: TBD (V2)
+- **Dependencies**: Decisions on Open Questions 1 and 4
+
+---
+
+## Testing Strategy
+
+### Unit Tests
+
+- Test requirement checking logic with mock commands
+- Test parameter parsing (--langs, --all)
+
+### Integration Tests
+
+- Test actual requirement checking with real commands
+- Test in environments with missing dependencies
+- Test across different operating systems (Windows, macOS, Linux)
+- Test with virtual environments (Python) and version managers (nvm, gvm)
+
+### Manual Testing
+
+**Test Scenarios**:
+
+1. **Clean Environment**: Run in fresh environment with no tools installed
+   - Expected: All requirements reported as missing with instructions
+
+2. **Partial Installation**: Install some but not all requirements
+   - Expected: Correctly identify which are installed vs. missing
+
+3. **Multi-Language**: Run with `--langs python,java`
+   - Expected: Check requirements for both languages, report combined results, and ensure no other languages were evaluated
+
+4. **Current Repo Detection**: Run without parameters in different language repos
+   - Expected: Automatically detect and check correct language
+
+5. **MCP Server Startup**: Start MCP server in each language repo
+   - Expected: Auto-verify requirements and inform user of any issues
+
+6. **Agent Interaction**: Ask agent "check my environment setup"
+   - Expected: Agent invokes tool and presents results naturally
+
+### Cross-Language Validation
+
+Test in each language repository:
+
+- [ ] azure-sdk-for-python
+- [ ] azure-sdk-for-java
+- [ ] azure-sdk-for-js
+- [ ] azure-sdk-for-net
+- [ ] azure-sdk-for-go
+- [ ] azure-rest-api-specs
+
+Verify:
+
+- Language detection works correctly
+- Requirements are appropriate for each language
+- Installation instructions are accurate and complete
+
+---
+
+## Related Links
+
+- [Tool Issue: VerifySetup tool for V1 · Issue #12287](https://github.com/Azure/azure-sdk-tools/issues/12287)
+- [azsdkcli Issue: verifySetup · Issue #11837](https://github.com/Azure/azure-sdk-tools/issues/11837)
+- [Python MCP Implementation Reference](https://github.com/Azure/azure-sdk-for-python/blob/ad3e09230bfbf31030c375102ecef366f4bcb35a/eng/tools/mcp/azure-sdk-python-mcp/main.py)
+
+---
+
+## Future Work
+
+- Determine how to dynamically get requirements, instead of relying on a hard-coded config
+- Explore invoking verifySetup when another tool fails due to missing dependencies
+
+## Meeting Notes
+
+### 10/9 Discussion - Key Takeaways
+
+1. **Source of Truth**: Agreement that there should be one source-of-truth config (from eng/common or each repo) for all requirements
+   - Consider devcontainers as potential source
```

Comment:
Caveat - you can't totally do this because devcontainers are Linux, and we'll see a lot of Windows developers. However, it's a start.